library/packages/src/lib/y2packager/product_reader.rb

```diff
@@ -145,21 +146,31 @@ def installed_base_product
       base = base_product
       return nil unless base
 
-      Y2Packager::Product.from_h(base)
+      Y2Packager::Product.new(
```

Comment:
You could have implemented a `Y2Packager::Product.from_resolvable` method, similar to `from_h`. That method should receive an additional `installation_package` argument too, which is not included in the resolvable.